### ISSUE TITLE
Got unexpected error without any clue how to work with it in CRA

### ISSUE THREAD

**Issue by vadimshvetsov** [TypeStrong/fork-ts-checker-webpack-plugin#661]
## Current behavior
npm run build
```
> my-app@0.2.3 build
> react-scripts build

Creating an optimized production build...
Failed to compile.

fork-ts-checker-webpack-plugin error in undefined(undefined,undefined):
Value undefined out of range for undefined options property undefined  TSINTERNAL
```
npm start
```
Files successfully emitted, waiting for typecheck results...
```

## Expected behavior
It should build without error and with `npm start` show compiled files

## Steps to reproduce the issue
I've just migrated code when update react-hook-form from v6 to v7. It's really hard to reproduce, this error message gives no clue where the issue occurs

## Issue reproduction repository
It's very hard to reproduce, looks like I need to make some mistake in code, but don't know which

## Environment
 - **react-scripts**: [4.0.3`]
 - **typescript**: [4.3.5]
 - **os**: [e.g. OS X Big Sur 11.5.2]

**Comment by vadimshvetsov** [comment]
I've found the reason of this bug. That was a nested type and it takes couple days to find that with removing files one by one from a codebase.

Who's using react-hook-form@v7 avoid approach like below on your integration components:
```
export const MultipleAutocompleteField = (props: MultipleAutocompleteFieldProps<T>) => {
  return (
    <Controller<ReactElement<MultipleAutocompleteFieldProps<T>>>
      ... 
    />
```

Sry for opening the issue here.
